diff --git a/node_modules/webpack/lib/library/UmdLibraryPlugin.js b/node_modules/webpack/lib/library/UmdLibraryPlugin.js
index b7013d7..ca52852 100644
--- a/node_modules/webpack/lib/library/UmdLibraryPlugin.js
+++ b/node_modules/webpack/lib/library/UmdLibraryPlugin.js
@@ -292,14 +292,25 @@ class UmdLibraryPlugin extends AbstractLibraryPlugin {
 						  externalsRequireArray("commonjs") +
 						  ");\n" +
 						  getAuxiliaryComment("root") +
-						  "	else\n" +
+						  "	else{\n" +
+						  "		var a = factory(" +
+						  externalsRootArray(externals) +
+						  ");\n" +
 						  "		" +
+						  replaceKeys(
+						    accessorAccess("root", names.root || names.commonjs)
+						  ) +
+						  " = " +
+						  replaceKeys(
+						    accessorAccess("root", names.root || names.commonjs)
+						  ) +
+						  " || {};\n" +
+						  "		for(var i in a) " +
 						  replaceKeys(
 								accessorAccess("root", names.root || names.commonjs)
 						  ) +
-						  " = factory(" +
-						  externalsRootArray(externals) +
-						  ");\n"
+						  "[i] = a[i];\n" +
+						  "	}\n"
 						: "	else {\n" +
 						  (externals.length > 0
 								? "		var a = typeof exports === 'object' ? factory(" +
